src/ImageSharp/Formats/Jpeg/Components/Decoder/ArithmeticScanDecoder.cs

```diff
@@ -326,11 +325,13 @@ private void ParseBaselineData()
 
             if (this.scanComponentCount != 1)
             {
+                this.spectralConverter.PrepareForDecoding();
```

Comment:
Okay, we got a little problem here. ITU specs allows to place markers between frame and scan markers which can screw up current architecture.

I've decided to separate SOF scan callback which now simply saves references to frame and jpeg data variables and has a separate `PrepareForDecoding()` callback which is called explicitely for single scan images and implicitely called by multi-scan images by `SpectralConverter`. `HuffmanScanDecoder` and `ArithmeticScanDecoder` are affected by this change.

I'm planning to do a refactoring PR for entire jpeg decoder after this is merged. Arithmetic decoding PR and this scaling decoding PR contains rather large changes so it's the perfect time to clean it up! :)